find_white_in_terminal.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python
+import sys
+import re
+
+
+in_prompt = False
+
+if len(sys.argv) != 2:
+    sys.stderr.write("ERROR: Usage %s <path to .tex file>\n", sys.argv[0])
+    sys.exit(1)
+
+tex_file = sys.argv[1]
+
+with open(tex_file) as infile:
+    for lineno, line in enumerate(infile.read().splitlines()):
+        if '\\begin{prompt}' in line:
+            in_prompt = True
+        if '%' in line and in_prompt:
+            # Find content between % %
+            latex_interpreted = re.findall(r'%([^%]*)%', line)
+            for match in latex_interpreted:
+                # Find latex command not ending in {}.
+                # Uses a negative lookahead (?!...) to assert that the latex
+                #  command doesn't end in a letter or {}
+                for latex_command in re.finditer(r'\\(\w+)(?![\w{])', match):
```

Comment:
@redfast00 What's the point of looking for `\w` in the negative lookahead? If there's a match for `\w`, it'll be consumed by the `\w+`?
Negative lookahead should be `(?!{})`?

Reply:
That was my first idea as well, but that didn't work: consider for example:
`%\shellcmd{whoami}`. The regex `\\(\w+)(?!{})` matches, because `shellcmd` has arguments (and isn't followed by `{}`).

The next logical thing to do would be to use a slightly weaker form: `\\(\w+)(?!{)`. That unfortunately also doesn't work on the example, since it will match `shellcm`, and since the last character (`d`) is not a `{`, it also matches the lookahead.

Reply:
OK, thanks for clarifying. :)